VDF.Core/MediaInfo.cs

```diff
@@ -64,6 +64,8 @@ public class StreamInfo {
 			[ProtoMember(11)]
 			public float FrameRate { get; set; }
 
+			[ProtoMember(12)]
+			public int Default { get; set; }
```

Comment:
This is a very bad name for a variable. Also can you explain why we need this?

Reply:
There seems to be something reserved `default` I noticed when I created the serialization class (but not jet read what's behind the default)

the problem to fix is to read the metadata of the one that ffmpeg displays / uses to create the thumbnails, if there is more than one video stream.

In the metadata of the stream there is the `disposition` block, in which there is a field `default` (furthermore also fields like `attached_pic`,..).
In the video examples I had seen, the relevant stream / the stream to be played was also flagged with default=1.
So my assumption was that ffmpeg also selects this stream if you don't specify one specifically.
=> My change reads the `default` flag, stores it in the `MediaInfo` (under the name `Default`) and uses the information to select the stream.

Reply:
But just now I came across the following page:
https://ffmpeg.org/ffmpeg.html#Automatic-stream-selection

Of course, I don't know if that's still current. Just as I don't know in which video container formats the disposition section exists... or however ffprobe creates it.

But back to the description I found: 

> 4.1.1 Automatic stream selection
> 
> In the absence of any map options for a particular output file, ffmpeg inspects the output format to check which type of streams can be included in it, viz. video, audio and/or subtitles. For each acceptable stream type, ffmpeg will pick one stream, when available, from among all the inputs.
> 
> It will select that stream based upon the following criteria:
> 
>     for video, it is the stream with the highest resolution,
>     for audio, it is the stream with the most channels,
>     for subtitles, it is the first subtitle stream found but there’s a caveat. The output format’s default subtitle encoder can be either text-based or image-based, and only a subtitle stream of the same type will be chosen. 
> 
> In the case where several streams of the same type rate equally, the stream with the lowest index is chosen. 

**Based on this, my extension to read the default flag would not be necessary because it is not the feature used by ffmpeg.** (second commit on this branch)

Instead, you would extend the first commit in this branch (use stream with lowest index) with the resolution criterion mentioned in the description and be done with it.
_(I will do this within the next days, its only a small local change... extended if condition + last resolution variable)_

This would then eliminate the need to use my modified JsonReader. Except of course, you like one of my two versions better.... But it would be also not bad, if not.

Reply:
This is very interesting, nice find!